File: php/src/Model/DocumentPart.php
<?php

declare(strict_types=1);

namespace Dokmatiq\DocGen\Model;

use Dokmatiq\DocGen\Internal\Serializer;

/** A single part in a multi-part composed document. */
final class DocumentPart
{
    /**
     * @param array<string, string>|null $fields
     */
    public function __construct(
        public readonly ?string $htmlContent = null,
        public readonly ?string $markdownContent = null,
        public readonly ?string $templateName = null,
        public readonly ?array $fields = null,
        public readonly ?PageSettings $pageSettings = null,
    ) {}

    /** @return array<string, mixed> */
    public function toArray(): array
    {
        return Serializer::toArray($this);
    }
}

File: java/src/main/java/com/dokmatiq/docgen/model/DocumentPart.java
package com.dokmatiq.docgen.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import java.util.List;
import java.util.Map;

/** A single part in a multi-part composed document. */
@JsonInclude(JsonInclude.Include.NON_NULL)
public record DocumentPart(
        String htmlContent,
        String markdownContent,
        String templateName,
        String templateBase64,
        Map<String, String> fields,
        Map<String, String> bookmarks,
        List<ImageData> images,
        List<QrCodeData> qrCodes,
        List<BarcodeData> barcodes,
        Map<String, TableData> tables,
        PageSettings pageSettings
) {
    public DocumentPart(String htmlContent) {
        this(htmlContent, null, null, null, null, null, null, null, null, null, null);
    }

    public DocumentPart(String htmlContent, String templateName) {
        this(htmlContent, null, templateName, null, null, null, null, null, null, null, null);
    }
}
